### PR TITLE
IDENTITY-6260: Fix when requesting claims from userinfo endpoint, some previously cached claims are returned irrespective of the scope.

### PR DESCRIPTION
**Pull request by hasinidilanka**

### CODE REVIEW

**Comment by hasinidilanka** [comment]
@isharak @omindu @mefarazath

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java [hunk 1]
```diff
@@ -66,7 +66,6 @@
     private static final String PHONE_NUMBER_VERIFIED = "phone_number_verified";
     private static final String EMAIL_VERIFIED = "email_verified";
     private static final String ADDRESS = "address";
-    Map<String, Object> claimsforAddressScope = new HashMap<>();
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java]: Is this instance variable something we introduced recently?

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java [hunk 2]
```diff
@@ -194,6 +194,8 @@ public String getResponseString(OAuth2TokenValidationResponseDTO tokenResponse)
 
         if (StringUtils.isNotEmpty(cacheEntry.getEssentialClaims())) {
             lstEssential = getEssentialClaims(cacheEntry.getEssentialClaims());
+        } else {
+            lstEssential = new ArrayList<>();
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java]: Can we have a meaningful name for this variable?
The current variable name 'lstEssential' does not reflect its purpose at least to me 😕

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java (hunk 1 above)
Comment (omindu) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java]: AFAIK this was introduced during the previous OIDC compliance effort.

File: components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java [hunk 3]
```diff
@@ -193,7 +191,9 @@ public String getResponseString(OAuth2TokenValidationResponseDTO tokenResponse)
         }
 
         if (StringUtils.isNotEmpty(cacheEntry.getEssentialClaims())) {
-            lstEssential = getEssentialClaims(cacheEntry.getEssentialClaims());
+            essentialClaims = getEssentialClaims(cacheEntry.getEssentialClaims());
+        } else {
+            essentialClaims = new ArrayList<>();
```
Comment (senthalan) [inline components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java]: essentialClaims is already initialized. Any specific reason to re-initialize here.
I think we can skip the class level initialization of essentialClaims and use this logic

Comment (hasinidilanka) [reply components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/impl/UserInfoJSONResponseBuilder.java]: Refactored in master.

**Pull request merged**